vault/build/titusvaults/sources/Vault.move

```diff
@@ -233,11 +340,153 @@ module titusvaults::Vault {
         event::emit(standard_withdraw_vault_event);
     }
 
+    // strick price
+    public (friend) fun setStrikePrice(_host: &signer, round_id: u64, new_strike_price: u64) acquires RoundState {
+        let host_addr = address_of(_host);
+        assert!(host_addr == @titusvaults, E_NOT_AUTHORIZED);
+
+        let state = borrow_global_mut<RoundState>(@titusvaults);
+
+        // assert we are during the deposit phase
+        let current_time = timestamp::now_microseconds();
+        let deposit_end_time = state.round_start_time + DEPOSIT_PHASE_DURATION;
+        assert!(current_time <= deposit_end_time, E_NOT_DEPOSIT_PHASE);
+
+        state.strike_price = new_strike_price;
+
+        // strike price updated event
+        let strike_price_updated_event = StrikePriceUpdatedEvent {
+            round_id: state.current_round_id,
+            strike_price: new_strike_price
+        };
+        event::emit(strike_price_updated_event);
+    }
+
+    // premium price 
+    public (friend) fun setPremiumPrice(_host: &signer, round_id: u64, new_premium_price: u64) acquires RoundState {
+        let host_addr = address_of(_host);
+        assert!(host_addr == @titusvaults, E_NOT_AUTHORIZED);
+
+        let state = borrow_global_mut<RoundState>(@titusvaults);
+
+        // assert we are during the deposit phase and active phase
+        let current_time = timestamp::now_microseconds();
+        let deposit_end_time = state.round_start_time + DEPOSIT_PHASE_DURATION;
+        let active_end_time = deposit_end_time + OPTION_EXPIRY_DURATION;
+        assert!(current_time >= state.round_start_time && current_time <= active_end_time, E_NOT_BETWEEN_IN_DEPOSIT_AND_ACTIVE_PHASE);
+
+        state.premium_price = new_premium_price;
+
+        // premium price updated event
+        let premium_price_updated_event = PremiumPriceUpdatedEvent {
+            round_id: state.current_round_id,
+            premium_price: new_premium_price
+        };
+        event::emit(premium_price_updated_event);
+    }
+
+    public (friend) fun executeRound<VaultT, AssetT>(_host: &signer, round_id: u64) acquires RoundState, VaultMap, Vault {
```

Comment:
borrow rounds from the vector using the round ID, instead of just borrowing the state. As in execute round, we will perform action on three different Round State structs.